examples/jsm/nodes/shadernode/ShaderNodeBaseElements.js

```diff
@@ -0,0 +1,266 @@
+import OperatorNode from '../math/OperatorNode.js';
+import CondNode from '../math/CondNode.js';
+
+// lights
+import ReflectedLightNode from '../lights/ReflectedLightNode.js';
+
+// utils
+import ArrayElementNode from '../utils/ArrayElementNode.js';
+import ConvertNode from '../utils/ConvertNode.js';
+
+// shader node utils
+import { ShaderNode, nodeObject, nodeObjects, nodeArray, nodeProxy, nodeImmutable, ConvertType, getConstNodeType, cacheMaps } from './ShaderNode.js';
+
+// shader node utils
+
+export { ShaderNode, nodeObject, nodeObjects, nodeArray, nodeProxy, nodeImmutable };
+
+export const color = new ConvertType( 'color' );
+
+export const float = new ConvertType( 'float', cacheMaps.float );
+export const int = new ConvertType( 'int', cacheMaps.int );
+export const uint = new ConvertType( 'uint', cacheMaps.uint );
+export const bool = new ConvertType( 'bool', cacheMaps.bool );
+
+export const vec2 = new ConvertType( 'vec2' );
+export const ivec2 = new ConvertType( 'ivec2' );
+export const uvec2 = new ConvertType( 'uvec2' );
+export const bvec2 = new ConvertType( 'bvec2' );
+
+export const vec3 = new ConvertType( 'vec3' );
+export const ivec3 = new ConvertType( 'ivec3' );
+export const uvec3 = new ConvertType( 'uvec3' );
+export const bvec3 = new ConvertType( 'bvec3' );
+
+export const vec4 = new ConvertType( 'vec4' );
+export const ivec4 = new ConvertType( 'ivec4' );
+export const uvec4 = new ConvertType( 'uvec4' );
+export const bvec4 = new ConvertType( 'bvec4' );
+
+export const mat3 = new ConvertType( 'mat3' );
+export const imat3 = new ConvertType( 'imat3' );
+export const umat3 = new ConvertType( 'umat3' );
+export const bmat3 = new ConvertType( 'bmat3' );
+
+export const mat4 = new ConvertType( 'mat4' );
+export const imat4 = new ConvertType( 'imat4' );
+export const umat4 = new ConvertType( 'umat4' );
+export const bmat4 = new ConvertType( 'bmat4' );
+
+// core
+
+// @TODO: ArrayUniformNode
+
+export const func = ( code ) => {
+
+	const node = nodeObject( new FunctionNode( code ) );
+
+	const call = node.call.bind( node );
+	node.call = ( params ) => nodeObject( call( params ) );
+
+	return node;
+
+};
+
+export const uniform = ( nodeOrType ) => {
+
+	const nodeType = getConstNodeType( nodeOrType );
+
+	// TODO: get ConstNode from .traverse() in the future
+	const value = nodeOrType.isNode === true ? nodeOrType.node?.value || nodeOrType.value : nodeOrType;
+
+	return nodeObject( new UniformNode( value, nodeType ) );
+
+};
+
+export const attribute = ( name, nodeOrType ) => nodeObject( new AttributeNode( name, getConstNodeType( nodeOrType ) ) );
+export const property = ( name, nodeOrType ) => nodeObject( new PropertyNode( name, getConstNodeType( nodeOrType ) ) );
+
+export const bypass = nodeProxy( BypassNode );
+export const code = nodeProxy( CodeNode );
+export const context = nodeProxy( ContextNode );
+export const expression = nodeProxy( ExpressionNode );
+export const functionCall = nodeProxy( FunctionCallNode );
```

Comment:
Let simplify here to `call`: 
`call( func, params )`

Reply:
Done.